### PR TITLE
Document when git tags are lightweight vs. annotated

### LINKED ISSUE #109: What kind of git tag bumpversion uses?
Hello there,

I think it is necessary to mention that you are using `Lightweight` tags in git. I couldn't find this information either in the document `README.md` or in the source code documentation. The only way to figure out is to read the source code!

Thanks,

### PR DESCRIPTION
**Pull request by GreatBahram**
Hello there,

Check this issue #109, please.

Closes #109

### CODE REVIEW

File: bumpversion/vcs.py [hunk 1]
```diff
@@ -126,9 +126,13 @@ def add_path(cls, path):
 
     @classmethod
     def tag(cls, sign, name, message):
+        """
```
Comment (florisla) [inline bumpversion/vcs.py]: The PEP8 recommendation is to start the docstring with a short, descriptive sentence (with leading capital and trailing dot).

The next line, if any, should be empty followed by paragraphs.

So first line could be like "Create a tag of the new_version in VCS."

Comment (GreatBahram) [reply bumpversion/vcs.py]: Thanks!

Comment (florisla) [reply bumpversion/vcs.py]: Still needs an empty line after the first one. Sorry ! :-)

Comment (GreatBahram) [reply bumpversion/vcs.py]: No, I am sorry!
Thanks!

**Comment by florisla** [comment]
@GreatBahram If you want to do further cleanup, you could describe the content of your changes in the title of this commit message.

E.g. call it `Document when git tags are lightweight vs. annotated`.

In the future, you could also put your changes in a branch with a descriptive name, e.g. `document-git-tag-type`.

File: README.md [hunk 2]
```diff
@@ -161,6 +161,11 @@ General configuration is grouped in a `[bumpversion]` section.
   Also available as command-line flag `tag-name`.  Example usage:  
   `bump2version --tag-name 'release-{new_version}' patch`
 
+  In addition, it is also possible to provide a tag message by using `--tag-message TAG_MESSAGE`. Example usage:
+  `bump2version --tag-name 'release-{new_version}' --tag-message "Release {new_version}" patch`
+
+  * If neither tag message or sign tag is provided, `bumpversion` uses a `lightweight` tag. Otherwise, it utilizes an `annotated` tag.
```
Comment (c4urself) [inline README.md]: Can you add: `see also: https://git-scm.com/book/en/v2/Git-Basics-Tagging`

Comment (ekohl) [reply README.md]: Is this also true for Mercurial?

Comment (GreatBahram) [reply README.md]: The link has been added, I am not sure about Mercurial, I don't think so.

**Comment by florisla** [comment]
Mercurial only has local (non-pushed) and regular tags.  See https://www.mercurial-scm.org/wiki/Tag .

**Comment by GreatBahram** [comment]
Thanks for the typo

**Pull request merged**